### ISSUE TITLE
App Directory: MDX & Link doesn't work properly with relative links

### ISSUE THREAD

**Issue by AnweshGangula** [vercel/next.js#50183]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
 Platform: win32
 Arch: x64
 Version: Windows 10 Pro
Binaries:
 Node: 16.13.0
 npm: N/A
warn  - Latest canary version not detected, detected: "13.4.3", newest: "13.4.4-canary.1".
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), MDX (@next/mdx), Routing (next/router, next/navigation, next/link)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/AnweshGangula/app-dir-mdx

### To Reproduce

I copied the code from [next.js/examples/app-dir-mdx](https://github.com/vercel/next.js/tree/canary/examples/app-dir-mdx). the only additional change I made was:

- create a new markdown file in [`app/blog/test/page.mdx`](https://github.com/AnweshGangula/app-dir-mdx/blob/master/app/blog/test/page.mdx)
- customized the built-in component for the[ HTML anchor tag `<a>`](https://github.com/AnweshGangula/app-dir-mdx/blob/master/mdx-components.tsx#L9) to use **`next/link`**

### Describe the Bug

in the `.mdx` pages, I have added relative links to one of the other pages in the folder like below:
- `[next page](./plain-markdown)`

I added the customization for the anchor tag in `mdx-components.tsx`. But after the customization, the links are not being redirected properly.

- when clicking on `http://localhost:3001/blog/test`, i'm being redirected to `http://localhost:3001/test`.

> Please note that the url in the DOM and even when I hover over the link show up as `http://localhost:3001/blog/test`

But what I noticed is that the url in the anchor tag href attribute is different when using app directory:

- before app directory: `<a href="/blog/plain-markdown">`
- after app directory: `<a href="./plain-markdown">` 

**Please note that this isue is caused on when I override the default `<a>` element to use the next/link component. with the default<`> element, it works fine.**

> I found [this article](https://nextjs.org/docs/messages/invalid-new-link-with-extra-anchor) which states that the next/link behaviour is changes in Next 13.
>
> Starting with Next.js 13, `<Link>` renders as `<a>`, so attempting to use `<a>` as a child is invalid.

### Expected Behavior

clicking on `http://localhost:3001/blog/test` should take me to the respective page

### Which browser are you using? (if relevant)

Microsoft Edge Version 113.0.1774.42 (Official build) (64-bit)

**Comment by joshtinhan** [comment]
@AnweshGangula The [next page](url) behavior in mdx file will redirect to the relative path. In your example, just add a dot in front of your path will solve the issue.
```mdx
<!-- //app/blog/test/page.mdx  -->
[next page](./hello-world) => [next page](../hello-world)
```

**Comment by AnweshGangula** [comment]
> just add a dot in front of your path will solve the issue.
> 
> ```gfm
> 
> [next page](./hello-world) => [next page](../hello-world)
> ```

This isn't correct.
if I'm in the URL `http://localhost:3000/blog/test` in my browser, then
- `[next page](./hello-world)` takes me to `http://localhost:3000/blog/hello-world`
- whereas `[next page](../hello-world)` takes me to `http://localhost:3000/hello-world`

**Comment by AnweshGangula** [comment]
This has been [resolved in Nextjs 13.4.5](https://github.com/vercel/next.js/pull/49521).
